File: 卡码/代码随想录/JS/链表/04-两两交换链表节点/LeetCode_0024.js
//24. 两两交换链表中的节点
function ListNode(val,next){
  this.val=(val===undefined ? 0 : val);
  this.next=(next===undefined ? null : next);
}

const swapPairs=head=>{
  //如果链表为空或者只有一个元素
  if(head==null||head.next==null) return head;
  let dummy=new ListNode(-1,head);
  let current=dummy;
  while(current.next&&current.next.next){
    let tmp=current.next; //保留1号位
    let tmp1=tmp.next.next; //保留3号位
    current.next=tmp.next;
    current.next.next=tmp;
    tmp.next=tmp1;
    current=tmp;
  }
  return dummy.next;
}

File: 卡码/代码随想录/Python/链表/04-两两交换链表节点/LeetCode_0024.py
#24. 两两交换链表中的节点
from typing import *

class ListNode:
	def __init__(self,val=0,next=None):
		self.val=val
		self.next=next

class Solution:
	def swapPairs(self,head:Optional[ListNode])->Optional[ListNode]:
		#如果链表为空或者只有一个元素
		if head==None or head.next==None:
			return head
		dummy=ListNode(-1,head)
		current=dummy
		while current.next and current.next.next:
			tmp=current.next	#保留1号位
			tmp1=tmp.next.next	#保留3号位
			current.next=tmp.next
			current.next.next=tmp
			tmp.next=tmp1
			current=tmp
		return dummy.next